### ISSUE TITLE
Couple of issues

### ISSUE THREAD

**Issue by Aminity** [MrHydralisk/Lethal-Company-Enhanced-Radar-Booster#9]
Hi, First off, I love the concept of this mod. Really love the idea and work you put into it.
I don't know if stuff I am describing below here is because of conflicting mods. But I still had some issues when turning all other mods off. I will mark with <> what I (still) had without other mods.

- When teleporting a radar booster into the facility, I was not able to teleport to said radar booster.
- <>When teleporting back a radar booster is undeployed.
- When trying to teleport a radar booster back, it didn't come back. (I don't know what other mod causes the issue, but here is my modlistcode if you want to look into it. 018d605a-7bac-bf91-13f7-de4ababafd10 ). It does work without other mods.
- <>some of the codes are not very readable on the 2x zoom. 
- <> Radar boosters that get left behind will be present on the next map.
- <> Radar Boosters that get teleported back to the ship become undeployed, refocussing the map on the terminal to what appears the center of the facility if there are other radar boosters still present.

Thanks in advance for taking a look at this. Good luck and hope you keep making cool mods.

**Comment by MrHydralisk** [comment]
Right away without additional info I can tell about:
2 - Is intended way. Same how you can teleport undeployed RB with Inverse Teleporter and it will be deployed.
4 - It understandable, after all game mini map wasn't made for such scale. There is config to adjust for smaller zoom, if it too big issue for you.
5 - That one is known thing, just haven't had time to change it, since it not a game breaking issue.
6 - That's how Vanilla game works even when you was watching RB on map and player or loot bug pick it up and undeployed.

**Comment by MrHydralisk** [comment]
For 1 and 3 I need to know what your config settings and what your logs show related to this mod or other errors in it.

**Comment by Aminity** [comment]
On the enhancedradarbooster mod I have everything set to default settings. If you would like to know them from all the other mods, I need to know if there is an easy way to share them all or have to refer you to the mod share code in the previous post.

**Comment by Aminity** [comment]
The BepInEX 5.4.21.0 log shown while running the game while using the inverse teleporter. I was teleported into the facility instead to the selected Radar Booster.

[Info   : Unity Log] Cooldown complete. Hiding UI.
[Info   : Unity Log] Triggering animated object trigger bool: setting to True
[Info   :SharedInverseTeleporter] Set teleport position to (53.99, -219.22, 59.95)
[Info   :SharedInverseTeleporter] Teleporting player 0 to (53.99, -219.22, 59.95)
[Info   : Unity Log] Teleporter shaking camera


[Info   : Unity Log] Cooldown complete. Hiding UI.
[Info   : Unity Log] Triggering animated object trigger bool: setting to True
[Info   :SharedInverseTeleporter] Set teleport position to (54.56, -212.97, 98.27)
[Info   : Unity Log] Current behaviour state: 1
[Info   : Unity Log] CHANGING BEHAVIOUR STATE!!! to 0
[Info   :SharedInverseTeleporter] Teleporting player 0 to (54.56, -212.97, 98.27)
[Info   : Unity Log] Teleporter shaking camera

I am geussing for question 1 it is conflicting with the mod Shared inverse teleporter?


For question 3 teleporting a radar booster back:

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
LCCollectYourDead.Patches.ShipTeleporterPatch.CashEmIn (ShipTeleporter __instance) (at <503c92985df94467b184a2921c91d692>:IL_0010)
(wrapper dynamic-method) ShipTeleporter.DMD<ShipTeleporter::beamUpPlayer>(ShipTeleporter)
ShipTeleporter.PressTeleportButtonClientRpc () (at <af9b1eec498a45aebd42601d6ab85015>:IL_0111)
ShipTeleporter.__rpc_handler_2773756087 (Unity.Netcode.NetworkBehaviour target, Unity.Netcode.FastBufferReader reader, Unity.Netcode.__RpcParams rpcParams) (at <af9b1eec498a45aebd42601d6ab85015>:IL_002E)
Unity.Netcode.RpcMessageHelpers.Handle (Unity.Netcode.NetworkContext& context, Unity.Netcode.RpcMetadata& metadata, Unity.Netcode.FastBufferReader& payload, Unity.Netcode.__RpcParams& rpcParams) (at <895801699cfc4b4ab52267f31e2a4998>:IL_004E)
Rethrow as Exception: Unhandled RPC exception!
UnityEngine.Debug:LogException(Exception)
Unity.Netcode.RpcMessageHelpers:Handle(NetworkContext&, RpcMetadata&, FastBufferReader&, __RpcParams&)
Unity.Netcode.ClientRpcMessage:Handle(NetworkContext&)
Unity.Netcode.NetworkBehaviour:__endSendClientRpc(FastBufferWriter&, UInt32, ClientRpcParams, RpcDelivery)
ShipTeleporter:PressTeleportButtonClientRpc()
ShipTeleporter:PressTeleportButtonServerRpc()
ShipTeleporter:__rpc_handler_389447712(NetworkBehaviour, FastBufferReader, __RpcParams)
Unity.Netcode.RpcMessageHelpers:Handle(NetworkContext&, RpcMetadata&, FastBufferReader&, __RpcParams&)
Unity.Netcode.ServerRpcMessage:Handle(NetworkContext&)
Unity.Netcode.NetworkBehaviour:__endSendServerRpc(FastBufferWriter&, UInt32, ServerRpcParams, RpcDelivery)
ShipTeleporter:PressTeleportButtonServerRpc()
ShipTeleporter:PressTeleportButtonOnLocalClient()
UnityEngine.Events.UnityEvent`1:Invoke(PlayerControllerB)
InteractTrigger:Interact(Transform)
GameNetcodeStuff.PlayerControllerB:Interact_performed(CallbackContext)
UnityEngine.InputSystem.LowLevel.<>c__DisplayClass7_0:<set_onUpdate>b__0(NativeInputUpdateType, NativeInputEventBuffer*)
UnityEngineInternal.Input.NativeInputSystem:NotifyUpdate(NativeInputUpdateType, IntPtr)

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
LCCollectYourDead.Patches.ShipTeleporterPatch.CashEmIn (ShipTeleporter __instance) (at <503c92985df94467b184a2921c91d692>:IL_0010)
(wrapper dynamic-method) ShipTeleporter.DMD<ShipTeleporter::beamUpPlayer>(ShipTeleporter)
ShipTeleporter.PressTeleportButtonClientRpc () (at <af9b1eec498a45aebd42601d6ab85015>:IL_0111)
ShipTeleporter.PressTeleportButtonServerRpc () (at <af9b1eec498a45aebd42601d6ab85015>:IL_00BB)
ShipTeleporter.__rpc_handler_389447712 (Unity.Netcode.NetworkBehaviour target, Unity.Netcode.FastBufferReader reader, Unity.Netcode.__RpcParams rpcParams) (at <af9b1eec498a45aebd42601d6ab85015>:IL_002E)
Unity.Netcode.RpcMessageHelpers.Handle (Unity.Netcode.NetworkContext& context, Unity.Netcode.RpcMetadata& metadata, Unity.Netcode.FastBufferReader& payload, Unity.Netcode.__RpcParams& rpcParams) (at <895801699cfc4b4ab52267f31e2a4998>:IL_004E)
Rethrow as Exception: Unhandled RPC exception!
UnityEngine.Debug:LogException(Exception)
Unity.Netcode.RpcMessageHelpers:Handle(NetworkContext&, RpcMetadata&, FastBufferReader&, __RpcParams&)
Unity.Netcode.ServerRpcMessage:Handle(NetworkContext&)
Unity.Netcode.NetworkBehaviour:__endSendServerRpc(FastBufferWriter&, UInt32, ServerRpcParams, RpcDelivery)
ShipTeleporter:PressTeleportButtonServerRpc()
ShipTeleporter:PressTeleportButtonOnLocalClient()
UnityEngine.Events.UnityEvent`1:Invoke(PlayerControllerB)
InteractTrigger:Interact(Transform)
GameNetcodeStuff.PlayerControllerB:Interact_performed(CallbackContext)
UnityEngine.InputSystem.LowLevel.<>c__DisplayClass7_0:<set_onUpdate>b__0(NativeInputUpdateType, NativeInputEventBuffer*)
UnityEngineInternal.Input.NativeInputSystem:NotifyUpdate(NativeInputUpdateType, IntPtr)

For question 3, is it conflicting with the collectyourdead mod?

**Comment by Aminity** [comment]
I can confirm everything works if I dissable those 2 mods mentioned. I did notice something else now. When using the inverse teleporter the lighting goes weird, its as if everything is extremely dark. I also tested this with only the EnhancedRadarBooster and BepInEx installed. Not sure what causes this. The logs don't say anything other then the following line:

[Info   : Unity Log] Teleporter shaking camera

**Comment by Aminity** [comment]
Screenshots:

https://imgur.com/a/kztQ2mQ
https://imgur.com/a/oXQ46Ay

**Comment by MrHydralisk** [comment]
1. I checked the code of SharedInverseTeleporter. It basically just override values of where to teleport player. So instead of RB coordinates it feed coordinates from that mod. It not a conflict between mods, it just overriding vanilla values after ERB, so it values end up the ones you see. 
3. Someone before was reporting that LCCollectYourDead was breaking things for them. I think it was like giving errors in BeamUpPlayer method before part with ERB Teleport RB code, so it was just never reaching ERB code. If they fix their errors, there shouldn't be any conflict between two mods itself from what I understand.
- Lighting thing happens when teleported to RB that located outside facility. That how Inverse Teleportation in vanilla work. It apply Dungeon shader on iTeleportation, cause didn't expect you to teleport to location that not inside facility. It was a very rare thing for players to do, wasn't game breaking and I decided to not fix it, cause it might end up breaking compatability with something.
